firmware/Sources/HAL/HAL_STS/hal_sts.c

```diff
@@ -0,0 +1,118 @@
+/**********************************************************************************/
+uint16_t devAddres = 0x4A; // Two options: 0x4A or 0x4B
+/**********************************************************************************/
+/*                        Definition of exported variables                        */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                      Definition of exported constant data                      */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Declaration of local function prototypes                    */
+/**********************************************************************************/
+/**
+ * @fn HAL_STS_result_t _I2cTransfer()
+ * @brief Transfer the data
+ * @return
+ * 		@ref HAL_STS_RESULT_SUCCESS if initialized correctly,
+ * 		@ref HAL_STS_RESULT_ERROR otherwise
+ */
+static HAL_STS_result_e _I2cTransmit(const uint16_t devAddr, const uint8_t *comand);
+
+/**
+ * @fn HAL_STS_result_t _I2cTransfer()
+ * @brief Transfer the data
+ * @return
+ * 		@ref HAL_STS_RESULT_SUCCESS if initialized correctly,
+ * 		@ref HAL_STS_RESULT_ERROR otherwise
+ */
+static HAL_STS_result_e _I2cReceive(const uint16_t devAddr, uint16_t *data );
+/**********************************************************************************/
+/*                       Definition of local constant data                        */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                         Definition of local functions                          */
+/**********************************************************************************/
+static HAL_STS_result_e _I2cTransmit(const uint16_t devAddr, const uint8_t *command){
+	uint16_t header = devAddr << 1 | 1;
+	// Warning suppression: the command variable is an input and should be constant.
+	HAL_STS_result_e res = HAL_I2C_Master_Transmit(&hi2c1, header, command, 2, EPC_CONF_I2C_TIMEOUT);
+	return res;
+}
+
+static HAL_STS_result_e _I2cReceive(const uint16_t devAddr, uint16_t *dataO ){
+	uint16_t header = devAddr << 1 | 1;
+	uint8_t data[4];
+	HAL_STS_result_e res = HAL_I2C_Master_Receive(&hi2c1, header, data, 3, EPC_CONF_I2C_TIMEOUT);
+	*dataO = data[0]<<8 | data[1];
+	return res;
+}
+/**********************************************************************************/
+/*                        Definition of exported functions                        */
+/**********************************************************************************/
+
+HAL_STS_result_e HAL_StsInit (void){
+	HAL_STS_result_e res = HAL_STS_RESULT_SUCCESS;
+	error_raised = 0;
```

Comment:
type¿? does this work?

Reply:
Your problem, not mine.

Reply:
include in .c ``include "epc_st_err.h" //Import EPC_ST_ERR_COUNTER`

and change `error_raised` by `EPC_ST_ERR_COUNTER`